### PR TITLE
FE: API, Code Cleanup & Fixes

### PR DESCRIPTION
**Pull request by benniloidl**

### CODE REVIEW

**Comment by Jozys** [comment]
Please provide a description of what you changed in this PR.

**Review by Integraluminium — CHANGES_REQUESTED** [review]
I think there are many changes, which are a bit confusing to look at.
Please check license for the pictures and export for i18n.

_Have you tested usability on firefox?_

File: frontend/src/components/avatar/Avatar.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import "./Avatar.scss";
+import React from "react";
+
+export default function Avatar(props: { userId?: string }) {
+	return <img className={ "avatar" }
+				src={ "" }
+				alt={ "Avatar" }
+				onError={ ({ currentTarget }) => {
+					currentTarget.onerror = null; // prevents looping
+					currentTarget.src = "https://www.gravatar.com/avatar/205e460b479e2e5b48aec07710c08d50";
```
[the remark this answers is not in the document]
Comment (benniloidl) [reply frontend/src/components/avatar/Avatar.tsx:10]: Yes

**Comment by benniloidl** [comment]
The requests have been implemented (even though they did not regard the changes in this PR).

**Comment by Integraluminium** [comment]
Looks good for me,
+ Internal server error is fixed in #227 due to errors in backend
+ The button to post answers is missing, please fix in next PR
+ I18n is provided partly, please move on and extract more elements into the language file

**Review by Integraluminium — APPROVED** [review]
Please review last comment

**Pull request merged**